src/Storages/Distributed/DirectoryMonitor.cpp

```diff
@@ -280,8 +279,11 @@ void StorageDistributedDirectoryMonitor::processFile(const std::string & file_pa
         Settings insert_settings;
         std::string insert_query;
         ClientInfo client_info;
+
         readHeader(in, insert_settings, insert_query, client_info, log);
 
+        auto connection = pool->get(timeouts, &insert_settings);
```

Comment:
Nice! And FWIW `dist_settings` can be removed from here:

https://github.com/ClickHouse/ClickHouse/blob/58786f992988a16a8ea787fe624c791529343702/tests/integration/test_insert_distributed_load_balancing/test.py#L37